modules/microshift-lvms-csi-snapshot-optional.adoc

```diff
@@ -0,0 +1,103 @@
+
+[id="lvms-stop-deployment_{context}"]
+== Stopping LVMS and CSI volume snapshot deployment
+
+Use the following procedure to prevent the deployment of LVMS CSI driver and CSI volume snapshot controller.
+
+.Procedure
+
+. To stop the LVMS CSI deployment, enter the following `driver` value under the `.storage` node in the {microshift-short} config file:
++
+[source,yaml]
+----
+# ...
+  storage
+   driver:'' <1>
+# ...
+----
+<1> Accepted values are `none` or `lvms`.
++
+[NOTE]
+====
+If the `driver` value is empty or null, {microshift-short} defaults to deploying LVMS.
+====
+
+. To stop CSI volume snapshot controller deployment, enter the following `optionalCsiComponents` value under the `.storage` node in the {microshift-short} config file:
++
+[source,yaml]
+----
+# ...
+  storage
+   optionalCsiComponents:'' <1>
+# ...
+----
+<1> Accepted values are `csi-snapshot-controller`, `csi-snapshot-webhook`, or `none`. `none` is mutually exclusive with all other values.
++
+[NOTE]
+====
+If the `optionalCsiComponents` value is empty or unspecified, {microshift-short} defaults to deploying snapshot-controller and snapshot-webhook.
+====
+
+. To restart {microshift-short} after the `.storage` node is specified with supported values in the {microshift-short} config file, run the following command:
++
+[source,terminal]
+----
+$ sudo systemctl restart microshift
+----
++
+[NOTE]
+====
+MicroShift does not redeploy the disabled components after restart.
+====
+
+[id="lvms-remove-installed-lvms_{context}"]
+== Removing installed LVMS and CSI volume snapshot
+
+To uninstall the deployed LVMS CSI driver and CSI volume snapshot controller, use the following procedure.
+
+.Procedure
+
+. To uninstall the LVMS CSI driver, run the following commands:
++
+[source,terminal]
+----
+$ oc delete -n openshift-storage daemonset.apps/vm-manager
+$ oc delete -n openshift-storage deployment.apps/lvms-operator
+$ oc delete storageclasses.storage.k8s.io/topolvm-provisioner
+----
++
+.Example output
+[source,terminal]
+----
+daemonset.apps "vg-manager" deleted
+deployment.apps "lvms-operator" deleted
+storageclass.storage.k8s.io "topolvm-provisioner" deleted
+----
+
+. To uninstall the CSI volume snapshot controller, run the following command:
++
+[source,terminal]
+----
+$ oc delete -n kube-system deployment.apps/csi-snapshot-controller deployment.apps/csi-snapshot-webhook
+----
++
+.Example output
+[source,terminal]
+----
+deployment.apps "csi-snapshot-controller" deleted
+deployment.apps "csi-snapshot-webhook" deleted
+----
++
+[IMPORTANT]
+====
+Automated un-installation is not supported as this can cause orphaning provisioned volumes. Without the LVMS CSI driver, the cluster has no knowledge of the underlying storage interface and thus cannot perform provisioning/deprovisioning or mount/unmount operations. Workloads with attached volumes must be manually stopped, and those volumes must then be manually deleted by the user.
```

Comment:
I think this note should go at the start of the section. At this point, the task has been done so informing a customer here might cause some annoyance on their part.

Note: 
![Screenshot from 2024-08-30 14-21-18](https://github.com/user-attachments/assets/88ca3e14-eba3-488c-bff7-bc7f3befa025)

Reply:
Workloads with attached volumes must be manually stopped, and those volumes must then be manually deleted by the user.

Maybe this can be removed or added to the procedure lead-in sentence? Might be self-explanatory given the instructions.

The IBM SSG has no issues with "thus" anymore, but I'm not a fan of it as it might be considered "stuffy" language. Perhaps replace with "so it"?

Reply:
done